posts/2020/11/Design In Open Source

```diff
@@ -0,0 +1,39 @@
+In my experience, communities have always been welcoming of design work, so this scenario usually ends well. Still, it exposes exactly how the value system that drives a lot of open source work (trust and reputation) fails people who do non-development work (not only design) in this field. Personally, I’m grateful that people tend to trust my word when I say I’m a designer or that there have always been developers in the communities I’ve worked with willing to vouch for me if needed. It still is extra time and work siphoning energy from the actual design, and continues a greater trend of lack of trackable credit for design work.
+## More on the mundane aspects of design in open-source
+None of this is to say that design contributions are not welcome in open-source, in fact I’ve personally found the opposite to be true. But at its core, open-source is, and has generally been, focussed on generating code. This focus is a choice, and this choice has impacts on the community, governance, direction, and more. Given the constraints that go into a decentralized team, often working on different priorities, it can be difficult to know in what direction the project is heading. This process often generates useful software that’s technically sound, after all, as Vincent Van Gogh said, “Great things are done by a series of small things being brought together.” Open-source does a wonderful job of improving large complex software systems by iterating on bite-sized pieces, but this work often comes in spurts when maintainers have time and inspiration.
+It can be hard to invest into a thorough design process if you’re not sure what’s going to be developed next, or ever. And if you do take that chance and invest in a design, there’s an uncomfortably high chance you’re missing the mark of what the maintainer thinks the next big steps of their project are (likely documented only in their mind). If members of a project don’t intentionally take time to prioritize design, it can be easier to just merge changes directly into the code base and see how the community likes it, as opposed to putting significant amounts of time into design. Many features that end up in open-source software are the result of one or two software developers writing the code and offering it directly to the project, while even the most well executed design still needs to be coded. It often happens that the solution that’s working now, is the priority over the solution that might work better, if it is even built.
+In my experience, it’s easier to design with a specific, prioritized, goal and user base in mind, than to tackle a giant web of problems that have varying degrees of dependency on one another. And without some consensus around what will be prioritized next, or who will follow through to implement things like a refreshed User Interface, it’s risky to commit enough for larger scale design projects. Which brings us to another interesting part of design.
```

Comment:
Replace paragraph.
```suggestion
**IPF:** Because features are often already or mostly built by the time they are revealed to the team (designers included), they have also already been designed—intentionally or not—by the people who built that feature. This process works against involving designer perspectives where they might be most helpful or impactful.
```